### DISCUSSION TITLE
Regarding reproducibility of the results.

### DISCUSSION THREAD

**Discussion by AdityaJainTD** [ContinualAI/avalanche#1067] [Q&A] [answered]
Hi,

I have been training with different strategies (EWC, Replay, Hybrid). Often times the evaluation accuracy ranges from 88.xx - 92.xx for the same code without any changes applied to it. I have provided the seed value for the nc_benchmark function. Do I need to seed anything else to generate reproducible results?

**Comment by AndreaCossu** [comment, marked as answer]
In addition, try calling this function at the beginning of your main

```python

def set_seed(seed):
    if seed is None:
        return
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    np.random.seed(seed)
    random.seed(seed)
    if torch.cuda.is_available():
        torch.backends.cudnn.deterministic = True
        torch.backends.cudnn.enabled = True
        torch.backends.cudnn.benchmark = False
```

**Reply by AdityaJainTD** [reply]
I tried adding this function into my code. I ran it quite a few times, my score was still fluctuating between 85.xx - 90.xx.

**Reply by AdityaJainTD** [reply]
Hi, 
I am still facing the same issue. Is there anything else that I need to add seed value to?

**Reply by AntonioCarta** [reply]
Avalanche doesn't really add any randomization mechanism. They all reuse underlying libraries (numpy, python random module, pytorch). You can check the [pytorch randomness guide](https://pytorch.org/docs/stable/notes/randomness.html) to see if there is anything missing.
